packages/network-controller/src/NetworkController.ts

```diff
@@ -991,201 +1293,627 @@ export class NetworkController extends BaseController<
+          )}`,
+        );
+      }
+
+      if (
+        networkClientId !== undefined &&
+        !Object.values(autoManagedNetworkClientRegistry).some(
+          (networkClientsById) => networkClientId in networkClientsById,
+        )
+      ) {
+        throw new Error(
+          `Cannot update network: RPC endpoint '${
+            // This is a string.
+            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+            url
+          }' refers to network client ${inspect(
+            networkClientId,
+          )} that does not exist`,
+        );
+      }
+
+      if (
+        setOfNewRpcEndpointFields.some(
+          (otherNewRpcEndpointFields) =>
+            otherNewRpcEndpointFields !== newRpcEndpointFields &&
+            URI.equal(otherNewRpcEndpointFields.url, newRpcEndpointFields.url),
+        )
+      ) {
+        throw new Error(
+          `Cannot update network: Each entry in rpcEndpoints must have a unique URL`,
+        );
+      }
+
+      for (const networkConfiguration of networkConfigurationsForOtherChains) {
+        const rpcEndpoint = networkConfiguration.rpcEndpoints.find(
+          (existingRpcEndpoint) =>
+            URI.equal(newRpcEndpointFields.url, existingRpcEndpoint.url),
+        );
+        if (rpcEndpoint) {
+          throw new Error(
+            `Cannot update network to point to same RPC endpoint as existing network for chain ${networkConfiguration.chainId} ('${networkConfiguration.name}')`,
+          );
+        }
+      }
+    }
+
+    if (
+      [...new Set(setOfNewRpcEndpointFields)].length <
+      setOfNewRpcEndpointFields.length
+    ) {
+      throw new Error(
+        'Cannot update network: Each entry in rpcEndpoints must be unique',
+      );
+    }
+
+    if ([...new Set(newNetworkClientIds)].length < newNetworkClientIds.length) {
+      throw new Error(
+        'Cannot update network: Each entry in rpcEndpoints must have a unique networkClientId',
+      );
+    }
+
+    if (infuraRpcEndpoints.length > 1) {
+      throw new Error(
+        'Cannot update network: There cannot be more than one Infura RPC endpoint',
+      );
+    }
+
+    const infuraRpcEndpoint = setOfNewRpcEndpointFields.find(
+      (newRpcEndpointFields) =>
+        newRpcEndpointFields.type === RpcEndpointType.Infura,
+    );
+    if (infuraRpcEndpoint) {
+      const infuraNetworkName = deriveInfuraNetworkNameFromRpcEndpointUrl(
+        infuraRpcEndpoint.url,
+      );
+      const infuraNetworkNickname = NetworkNickname[infuraNetworkName];
+      const infuraChainId = ChainId[infuraNetworkName];
+      if (newChainId !== infuraChainId) {
+        throw new Error(
+          // This is a string.
+          // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+          `Cannot update network with chain ID ${newChainId} and Infura RPC endpoint for '${infuraNetworkNickname}' which represents ${infuraChainId}, as the two conflict`,
+        );
```

Comment:
Not seeing an error here.

```suggestion
        throw new Error(
          `Cannot update network with chain ID ${newChainId} and Infura RPC endpoint for '${infuraNetworkNickname}' which represents ${infuraChainId}, as the two conflict`,
        );
```